internal/app/backend/backend_service.go

```diff
@@ -95,59 +101,64 @@ func (s *backendService) FetchMatches(req *pb.FetchMatchesRequest, stream pb.Bac
 	}
 
 	ctx := stream.Context()
-	matchChan := make(chan *pb.Match)
-	errChan := make(chan error)
+	resultChan := make(chan mmfResult)
 
-	go func(matchChan chan<- *pb.Match, errChan chan<- error) {
+	go func(resultChan chan<- mmfResult) {
 		var wg sync.WaitGroup
+		var matches []*pb.Match
 
 		defer func() {
 			wg.Wait()
-			close(matchChan)
-			close(errChan)
+			close(resultChan)
 		}()
 
 		for _, profile := range req.Profile {
 			wg.Add(1)
 			go func(profile *pb.MatchProfile) {
 				defer wg.Done()
 
-				// Get the channel over which the generated match results will be sent.
+				// Get the match results that will be sent.
+				// TODO: The matches returned by the MatchFunction will be sent to the
+				// Evaluator to select results. Until the evaluator is implemented,
+				// we channel all matches as accepted results.
 				switch (req.Config.Type).(type) {
 				case *pb.FunctionConfig_Grpc:
-					matchesFromGRPCMMF(ctx, profile, grpcClient, matchChan, errChan)
+					matches, err = matchesFromGRPCMMF(ctx, profile, grpcClient)
 				case *pb.FunctionConfig_Rest:
-					matchesFromHTTPMMF(ctx, profile, httpClient, baseURL, matchChan, errChan)
+					matches, err = matchesFromHTTPMMF(ctx, profile, httpClient, baseURL)
 				}
+
+				resultChan <- mmfResult{matches, err}
 			}(profile)
 		}
-	}(matchChan, errChan)
+	}(resultChan)
 
-	for {
+	var done bool
```

Comment:
Remove this done logic, and just return ctx.Err()
It only returns non-nil if it was canceled, so it gets the same behavior without the bookkeeping.